### PR TITLE
Fix 'Directory.Packages.props' import logic

### LINKED ISSUE #9841: [CPVM] Cannot import packages file from a custom directory
## Problem

The '_DirectoryPackagesPropsBasePath' property is not respected when importing 'Directory.Packages.props'. Thus, We can't import the central packages file from a custom directory.

## Repro

1. Create any project with CPVM enabled (i.e. with `Directory.Packages.props`)

2. Override `_DirectoryPackagesPropsBasePath` with the directory of `Directory.Packages.props` anywhere you prefer.

3. Build with binlog `-bl`. e.g.: `dotnet build -bl` (_you can use [MSBuild Log Viewer](//msbuildlog.com) to inspect the binlog_)

4. Observe that the file is not imported.

### Sample Project and Verbose Logs

[NuGet-CPVM-Repro.zip](https://github.com/NuGet/Home/files/4968726/NuGet-CPVM-Repro.zip)

## Enviroment

NuGet product used: dotnet.exe nuget

NuGet version: 5.7.0.7

dotnet.exe --version: 5.0.100-preview.7.20366.6

VS version: 16.6.5

OS version: 10.0.19041.329

### PR DESCRIPTION
**Pull request by Nirmal4G**
## Bug

The '_DirectoryPackagesPropsBasePath' property is not used in the import logic. When we do override it, the property is not respected. Thus we can't import the packages file from a custom directory.

Fixes: NuGet/Home#9841
Regression: No

## Fix

Details: The import logic is same as the 'Directory.Build.props' logic present in the common props. The fix is to match their logic. That's all!

> I have also formatted the props/targets to follow EditorConfig rules. I've also made sure that `git blame` doesn't break with these changes.
> If you want to take the formatting changes into a separate PR or drop them, please let me know.

## Testing/Validation

Tests Added: No  
Reason for not adding tests:  It's same as the `Directory.Build.props` logic. Will add if we need them.
Validation:  Refer MSBuild's Common props

### CODE REVIEW

**Comment by zivkan** [comment]
@Nirmal4G please keep our PR template, including linking an issue in NuGet/Home, or creating a new issue if you can't find an appropriate existing issue. If this PR gets merged, the issue is used in our release notes.

Is there any customer impact? From my reading, it looks like it should be 100% functionally identical


@cristinamanum this appears to be related to how CPVM finds the Directory.Packages.props file. Does it look ok to you?

**Comment by Nirmal4G** [comment]
@zivkan @cristinamanum I've updated the PR description and created the issue.

**Comment by nkolev92** [comment]
Given that there's no functional benefit, I think we should not take this PR.
Especially considering the changes are mostly in the msbuild files. 

I believe `git blame` to be helpful command during development and as such it should remain useful rather than be littered with formatting changes. 

That's not to say, never fix formatting, rather say rarely fix the formatting.

**Comment by Nirmal4G** [comment]
So, Should I remove the formatting changes from this PR? You can see that I've segregated the changes.

> Given that there's no functional benefit

The last commit is the functional benefit. It fixes the import logic for `Directory.Packages.props`.

**Comment by Nirmal4G** [comment]
> That's not to say, never fix formatting, rather say rarely fix the formatting.

Those files are so messed up, this is what a rarely fixed formatting looks like. 😅😂

**Comment by Nirmal4G** [comment]
🔔 CI Please

**Comment by Nirmal4G** [comment]
@zivkan @nkolev92

**Comment by Nirmal4G** [comment]
🔔 CI Please

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets [hunk 1]
```diff
@@ -468,47 +475,48 @@ Copyright (c) .NET Foundation. All rights reserved.
...
-        <BuildAction Condition = "'%(AndroidResource.BuildAction)' == ''">AndroidResource</BuildAction>
+        <BuildAction Condition="'%(AndroidResource.BuildAction)' == ''">AndroidResource</BuildAction>
       </_PackageFiles>
       <_PackageFiles Include="@(BundleResource)" Condition=" %(BundleResource.Pack) == 'true' ">
-        <BuildAction Condition = "'%(BundleResource.BuildAction)' == ''">BundleResource</BuildAction>
+        <BuildAction Condition="'%(BundleResource.BuildAction)' == ''">BundleResource</BuildAction>
       </_PackageFiles>
     </ItemGroup>
   </Target>
+
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets:521]: Please comment why these changes are needed. If  not needed please revert.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 2]
```diff
@@ -10,22 +10,32 @@ Copyright (c) .NET Foundation. All rights reserved.
 ***********************************************************************************************
 -->
 
-<Project>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:13]: Namespace not needed.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 3]
```diff
@@ -10,22 +10,32 @@ Copyright (c) .NET Foundation. All rights reserved.
...
     <_DirectoryPackagesPropsFile Condition="'$(_DirectoryPackagesPropsFile)' == ''">Directory.Packages.props</_DirectoryPackagesPropsFile>
-    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
-    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::GetPathOfFileAbove('$(_DirectoryPackagesPropsFile)', '$(MSBuildProjectDirectory)'))</DirectoryPackagesPropsPath>
+    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove('$(MSBuildProjectDirectory)', '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
+    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::NormalizePath('$(_DirectoryPackagesPropsBasePath)', '$(_DirectoryPackagesPropsFile)'))</DirectoryPackagesPropsPath>
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:32]: Good catch!

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props [hunk 4]
```diff
@@ -10,22 +10,32 @@ Copyright (c) .NET Foundation. All rights reserved.
...
+      they did not already specify an absolute path to use via $(DirectoryPackagesPropsPath)
+  -->
   <PropertyGroup Condition="'$(ImportDirectoryPackagesProps)' == 'true' and '$(DirectoryPackagesPropsPath)' == ''">
     <_DirectoryPackagesPropsFile Condition="'$(_DirectoryPackagesPropsFile)' == ''">Directory.Packages.props</_DirectoryPackagesPropsFile>
-    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
-    <DirectoryPackagesPropsPath Condition="'$(_DirectoryPackagesPropsBasePath)' != '' and '$(_DirectoryPackagesPropsFile)' != ''">$([MSBuild]::GetPathOfFileAbove('$(_DirectoryPackagesPropsFile)', '$(MSBuildProjectDirectory)'))</DirectoryPackagesPropsPath>
+    <_DirectoryPackagesPropsBasePath Condition="'$(_DirectoryPackagesPropsBasePath)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove('$(MSBuildProjectDirectory)', '$(_DirectoryPackagesPropsFile)'))</_DirectoryPackagesPropsBasePath>
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:31]: Please comment what changed from the previous version. If nothing please revert.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets [hunk 5]
```diff
@@ -1144,7 +1145,7 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <!-- Remove projects that do not support restore. -->
     <!-- With SkipNonexistentTargets -->
-    <MsBuild
+    <MSBuild
```
Comment (cristinamanum) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:1148]: Why this change? (And similar above and below)

**Comment by cristinamanum** [comment]
> this appears to be related to how CPVM finds the Directory.Packages.props file. Does it look ok to you?

The line 32 in NuGet.props is the fix. The other changes seem to be style fixes.

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets:521]: +1 here. 
We prefer not having too many changes that don't have a functional benefit.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets [hunk 6]
```diff
@@ -1230,7 +1231,7 @@ Copyright (c) .NET Foundation. All rights reserved.
       <Output
           TaskParameter="TargetOutputs"
           PropertyName="_RestoreFallbackFoldersOverride" />
-    </MsBuild>
+    </MSBuild>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:1234]: Similar feedback as above, I think we should revert this.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 2 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:13]: Sure, but full framework MSBuild will complain if there is no namespace. Can you confirm that this props ships/imports only with Core MSBuild?

**Comment by Nirmal4G** [comment]
> Responding to Comments on formatting

Making these files readable with little formatting as possible.
- Fixing unnecessary spaces reduces cognitive load.
- Having new lines between tags and each section provides a clear cut and improves readability.
- Clear casing does contribute to above reasons but also prevents any mistakes in the future like the `WinFx`, `.Net`, `.dotNet` fiasco!!

I have separated the formatting changes and directory packages change into new PRs. I'll close this one.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets (hunk 6 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:1234]: Response: https://github.com/NuGet/NuGet.Client/pull/3538#issuecomment-668995259

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets (hunk 5 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets:1148]: Response: https://github.com/NuGet/NuGet.Client/pull/3538#issuecomment-668995259

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 4 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:31]: I did, in the commit, issue and the PR descriptions.

File: src/NuGet.Core/NuGet.Build.Tasks/NuGet.props (hunk 3 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks/NuGet.props:32]: Finally, Thanks for noticing!!

File: src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets (hunk 1 above)
Comment (Nirmal4G) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets:521]: Response: https://github.com/NuGet/NuGet.Client/pull/3538#issuecomment-668995259

**Comment by Nirmal4G** [comment]
Superseded by #3565 and #3566
